frontend/packages/client/src/components/CommunityCreate/StepOne.js

```diff
@@ -0,0 +1,272 @@
+import { WrapperResponsive, Dropdown } from "components";
+import { CommunityLinksForm } from "components/Community/CommunityEditorLinks";
+import useLinkValidator, {
+  urlPatternValidation,
+} from "../Community/hooks/useLinkValidator";
+import { getReducedImg } from "utils";
+import { useErrorHandlerContext } from "contexts/ErrorHandler";
+import { MAX_FILE_SIZE } from "const";
+import { pick } from "lodash";
+import { useCommunityCategory } from "hooks";
+
+const linksFields = [
+  "websiteUrl",
+  "twitterUrl",
+  "instagramUrl",
+  "discordUrl",
+  "githubUrl",
+];
+
+const initialValues = Object.assign(
+  {},
+  ...linksFields.map((key) => ({ [key]: "" }))
+);
+export default function StepOne({
+  stepData,
+  setStepValid,
+  onDataChange,
+  moveToNextStep,
+  isStepValid,
+}) {
+  const { notifyError } = useErrorHandlerContext();
+
+  const { data: communityCategory } = useCommunityCategory();
+
+  const setData = (data) => {
+    onDataChange(data);
+  };
+
+  const onDrop = useCallback(
+    (acceptedFiles) => {
+      acceptedFiles.forEach((imageFile) => {
+        // validate type
+        if (
+          !["image/png", "image/jpeg", "image/jpg"].includes(imageFile.type)
+        ) {
+          notifyError({
+            status: "Image Type not supported",
+            statusText: "Please upload a .png or .jpeg file type extension",
+          });
+          return;
+        }
+        // validate size
+        if (imageFile.size > MAX_FILE_SIZE) {
+          notifyError({
+            status: "Image file size not allowed",
+            statusText: "Please upload a new file (smaller than 5mb)",
+          });
+          return;
+        }
+        const imageAsURL = URL.createObjectURL(imageFile);
+
+        const img = new Image();
+        img.onload = function (e) {
+          // reduce image if necessary before upload
+          if (e.target.width > 150) {
+            getReducedImg(e.target, 150, "community_image").then((result) => {
+              onDataChange({
+                logo: { imageUrl: imageAsURL, file: result.imageFile },
+              });
+            });
+          } else {
+            onDataChange({ logo: { imageUrl: imageAsURL, file: imageFile } });
+          }
+        };
+        img.src = imageAsURL;
+      });
+    },
+    [onDataChange, notifyError]
+  );
+
+  const { getRootProps, getInputProps } = useDropzone({
+    onDrop,
+    maxFiles: 1,
+    accept: "image/jpeg,image/png",
+  });
+
+  const { communityName, communityDescription, logo, communityTerms } =
+    stepData || {};
+
+  // handle links form
+  const linksFieldsObj = Object.assign(
+    {},
+    initialValues,
+    pick(stepData || {}, linksFields)
+  );
+
+  const changeHandlers = useMemo(
+    () =>
+      Object.assign(
+        {},
+        ...linksFields.map((field) => ({
+          [field]: (value) =>
+            onDataChange({
+              [field]: value,
+            }),
+        }))
+      ),
+    [onDataChange]
+  );
```

Comment:
Would it work to use a closure pattern here? It appears that we are just generating an object based on the fields that all invoke the same function to update the data in the parent component.

something like this:
```js
// StepOne.js
const changeHandler = (field) => (value) => onDataChange({ [field]: value })

// CommunityEditorLinks.js
const Component = ({ onFieldChange }) => {
   return (
        <input 
            onChange={(event) =>
                onFieldChange("websiteUrl")(event.target.value)
          }
      />
```

Reply:
makes sense, updated the function as suggested